kong/tools/wrpc/threads.lua

```diff
@@ -76,7 +76,7 @@ local function step(wrpc_peer)
     msg, err = wrpc_peer:receive()
   end
 
-  if err ~= nil and not endswith(err, ": timeout") then
+  if err ~= nil and not endswith(err, ": timeout") and not endswith(err, ": closed") then
```

Comment:
Should this logic use a `wrpc_peer.closing` check instead? With a graceful shutdown we should receive a close frame, and therefore `wrpc_peer.closing` will be set `true`.

```suggestion
  if err ~= nil
    and not endswith(err, ": timeout")
    and not wrpc_peer.closing
  then
```

I'm not sure if I have it correct, but ideally we still log when an unexpected connection closure happens. A test case would look something like....

```
it("logs an error when the connection is terminated unexpectedly", function ()
  helpers.clean_logfile("servroot2/logs/error.log")
  assert(helpers.signal("servroot1", "-KILL"))
  need_exit = false
  assert.logfile("servroot2/logs/error.log").has.line("some kind of error string")
end)
```

Reply:
> Should this logic use a `wrpc_peer.closing` check instead? With a graceful shutdown we should receive a close frame, and therefore `wrpc_peer.closing` will be set `true`.
> 
> I'm not sure if I have it correct, but ideally we still log when an unexpected connection closure happens. A test case would look something like....
> 
> ```
> it("logs an error when the connection is terminated unexpectedly", function ()
>   helpers.clean_logfile("servroot2/logs/error.log")
>   assert(helpers.signal("servroot1", "-KILL"))
>   need_exit = false
>   assert.logfile("servroot2/logs/error.log").has.line("some kind of error string")
> end)
> ```

This is intentionally added. When we receive the closing frame, there could be an RPC call already fired, and blocking at `receive()`. It will see the error before the flag is set to closing. Of course, we should integrate the `close` handling with the `receive()` call. But this is just a compromise before we have a thorough refactoring.

And it seems we do not define details on what happens to a disconnected wRPC peer, like should it responds to all RPC calls not answered before closing, or send an error msg indicating those RPC calls are given up.

Reply:
Sounds reasonable enough for me. Thanks!